Libraries/CommonFunctions.psm1

```diff
@@ -2407,4 +2377,25 @@ Function Get-ExpectedDevicesCount {
     }
 
     return $expectedCount,$keyWord
+}
+
+# Extract values from file between two lines which has specified start pattern
+Function ExtractSSHPublicKeyFromPPKFile {
+    param (
+        [String] $filePath,
+        [String] $startLinePattern="Public-Lines",
+        [String] $endLinePattern="private-Lines"
+    )
+        if (![string]::IsNullOrEmpty($filePath)) {
+            $fromHereStartingLine = Select-String $filePath -pattern $startLinePattern | Select-Object LineNumber
+            $uptoHereStartingLine = Select-String $filePath -pattern $endLinePattern | Select-Object LineNumber
+            $extractedValue = ""
+            for($i=$fromHereStartingLine.LineNumber; $i -lt $uptoHereStartingLine.LineNumber-1; $i+=1) {
+                $extractedValue += Get-Content -Path $FilePath | Foreach-Object { ($_  -replace "`r*`n*","") } | Select-Object -Index $i
+            }
+
+            return "ssh-rsa $extractedValue"
+        } else {
+            return $null
```

Comment:
This is related to another my comment, this will derive another run-time error. Return null is ok, but we should get aways from landing on this block as much as possible.